### DISCUSSION TITLE
Multiple LDAP error

### DISCUSSION THREAD

**Discussion by xRemoN** [netbox-community/netbox#9360] [Q&A]
Trying to configure authorization for two MS AD domains in accordance with the official instruction - https://django-auth-ldap.readthedocs.io/en/latest/multiconfig.html. 
When I try to login, I get an error:
```
<class 'django.core.exceptions.ImproperlyConfigured'>
AUTH_LDAP_USER_SEARCH must be an LDAPSearch instance.
Python version: 3.8.10
NetBox version: 3.2.2
```
I get this error even if all search and group strings are commented out... 
I carefully checked the settings, I can't understand what the problem is.

My settings:
1. authentication.py - added to the bottom
```
# netbox.ldap
from django_auth_ldap.backend import LDAPBackend
class LDAPBackend1(LDAPBackend):
    settings_prefix = "AUTH_LDAP_1_"
class LDAPBackend2(LDAPBackend):
    settings_prefix = "AUTH_LDAP_2_"

```
2. ldap_config.py
```
import ldap
AUTH_LDAP_1_SERVER_URI = "ldaps://dc1.domain1.com"
AUTH_LDAP_1_USER_DN_TEMPLATE = "uid=%(user)s,OU=Users,OU=Accounts,DC=domain1,DC=com"

AUTH_LDAP_2_SERVER_URI = "ldaps://dc1.domain2.com"
AUTH_LDAP_2_USER_DN_TEMPLATE = "uid=%(user)s,OU=Users,DC=domain2,DC=com"

AUTHENTICATION_BACKENDS = ("netbox.ldap.LDAPBackend1", "netbox.ldap.LDAPBackend2")

< then duplicated everything AUTH_LDAP_*  to AUTH_LDAP_1_* and AUTH_LDAP_2_* >
AUTH_LDAP_1_CONNECTION_OPTIONS = {ldap.OPT_REFERRALS: 0}
AUTH_LDAP_2_CONNECTION_OPTIONS = {ldap.OPT_REFERRALS: 0}

AUTH_LDAP_1_BIND_DN = "CN=netboxacc,OU=Accounts,DC=domain1,DC=com"
AUTH_LDAP_1_BIND_PASSWORD = "Pwd1"
AUTH_LDAP_2_BIND_DN = "CN=netboxacc,OU=Users,DC=domain2,DC=com"
AUTH_LDAP_2_BIND_PASSWORD = "Pwd2"
...
```
3. configuration.py without any change 
```
REMOTE_AUTH_ENABLED = True
REMOTE_AUTH_BACKEND = 'netbox.authentication.LDAPBackend' 

```
REMOTE_AUTH_ENABLED = False also tried

4.	With one LDAP server, everything works great.

I am not a Python guru, can somebody help me out?

**Comment by annibac** [comment]
Hello,

Did you manage to solve your issue ? I'm facing the exact same one here ..

Thanks

**Reply by stoms12** [reply]
@jeremystretch  can you provide some guidance here. We have the same issue

**Reply by IdRatherStand** [reply]
I'm also in exactly the same position, we would like to have two different AD sources.
